### PR TITLE
[Enhancement] Imporve cross join cost penalty

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

In the current cross join cost model, the penalty work for right table memory cost.
When the left table is a large table and the right table is a small table, the penalty of cross join may become smaller

And in the plan of cross join with large table, the penalty of `100000000` is too small compared to the output of cross join

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by murphyatwork — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 25 / 27 (92.59%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Statistics.java|1|2|50.00%|[130]|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|7|8|87.50%|[285]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/JoinReorderLeftDeep.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/JoinReorderGreedy.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/JoinOrder.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/JoinReorderDP.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsEstimateCoefficient.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/cost/CostModel.java|6|6|100.00%|[]|

File: fe/fe-core/src/test/resources/sql/enumerate-plan/tpch-q7.sql [hunk 1]
```diff
@@ -39,73 +39,74 @@ order by
...
-                                SCAN (columns[45: N_NATIONKEY, 46: N_NAME] predicate[46: N_NAME IN (CANADA, IRAN)])
+                    INNER JOIN (join-predicate [36: C_CUSTKEY = 27: O_CUSTKEY AND 39: C_NATIONKEY = 50: N_NATIONKEY] post-join-predicate [null])
+                        SCAN (columns[36: C_CUSTKEY, 39: C_NATIONKEY] predicate[null])
                         EXCHANGE BROADCAST
-                            SCAN (columns[50: N_NATIONKEY, 51: N_NAME] predicate[51: N_NAME IN (IRAN, CANADA)])
+                            INNER JOIN (join-predicate [26: O_ORDERKEY = 9: L_ORDERKEY] post-join-predicate [null])
+                                SCAN (columns[26: O_ORDERKEY, 27: O_CUSTKEY] predicate[null])
+                                EXCHANGE BROADCAST
+                                    INNER JOIN (join-predicate [11: L_SUPPKEY = 1: S_SUPPKEY] post-join-predicate [null])
+                                        SCAN (columns[19: L_SHIPDATE, 9: L_ORDERKEY, 11: L_SUPPKEY, 14: L_EXTENDEDPRICE, 15: L_DISCOUNT] predicate[19: L_SHIPDATE >= 1995-01-01 AND 19: L_SHIPDATE <= 1996-12-31])
```
Comment (packy92) [inline fe/fe-core/src/test/resources/sql/enumerate-plan/tpch-q7.sql:56]: need test the performance of the sql.

Comment (Seaven) [reply fe/fe-core/src/test/resources/sql/enumerate-plan/tpch-q7.sql:56]: I test, the result of cross join only 2 rows, no impact on performance

**Review by packy92 — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by liuyehcf — APPROVED** [review]

**Comment by Seaven** [comment]
run starrocks_admit_test

**Pull request merged**